python/kvikio/_lib/libnvcomp_ll.pyx

```diff
@@ -407,6 +387,23 @@ class nvCompBatchAlgorithmLZ4(nvCompBatchAlgorithm):
         self.options = nvcompBatchedLZ4Opts_t(data_type)
         self.has_header = has_header
 
+        # CUDA kernel that copies uncompressed chunk size from the chunk header.
+        self._get_size_from_header_kernel = cupy.ElementwiseKernel(
+            "uint64 comp_chunk_ptr",
+            "uint64 uncomp_chunk_size",
+            "uncomp_chunk_size = *((unsigned int *)comp_chunk_ptr)",
```

Comment:
I have to use 32-bit pointer because that's how the header data is stored in [numcodecs LZ4 codec](https://github.com/zarr-developers/numcodecs/blob/cb155432e36536e17a2d054c8c24b7bf6f4a7347/numcodecs/lz4.pyx#L89). The header is just a `uint32_t` field that stores original, uncompressed data size. Unfortunately, that makes the header structure essentially frozen - there is no versioning or metadata info which would allow extending the header in the future, if needed.
I agree it may confuse readers, so I'm open to any suggestions to make this code more readable (add a comment?). Agree that ideally should also use fixed width `uint32_t` type instead of `unsigned int` (integer types are confusing in C++ standard), but, as you mentioned, it's not currently possible in cupy.

Reply:
Thanks! I think I was confused because I was trying to find the allocation of `comp_chunk_ptr` and I _thought_ it was an allocation uint64 entries, but I probably looked in the wrong place. So, 
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Perhaps add a comment here indicating which types these have to match up with?

Reply:
Done.